### PR TITLE
[Kotlin Cleanup] DirectGregorianInstantiation

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
[Kotlin Cleanup] DirectGregorianInstantiation

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by Akshay0701 — CHANGES_REQUESTED** [review]
Changes look good to me, but you forgot to remove the kotlinCleanup lines,
anyway, Thanks for your contributions!

File: lint-rules/src/main/java/com/ichi2/anki/lint/rules/DirectGregorianInstantiation.kt [hunk 1]
```diff
@@ -39,14 +37,14 @@ class DirectGregorianInstantiation : Detector(), SourceCodeScanner {
         )
     }
 
-    override fun getApplicableMethodNames(): List<String>? {
-        val forbiddenMethods: MutableList<String> = ArrayList()
+    override fun getApplicableMethodNames(): List<String> {
+        val forbiddenMethods: MutableList<String> = mutableListOf()
         forbiddenMethods.add("from")
         return forbiddenMethods
```
Comment (david-allison) [inline lint-rules/src/main/java/com/ichi2/anki/lint/rules/DirectGregorianInstantiation.kt:43]:
```suggestion
        return mutableListOf("from")
```

File: lint-rules/src/main/java/com/ichi2/anki/lint/rules/DirectGregorianInstantiation.kt [hunk 2]
```diff
@@ -39,14 +37,14 @@ class DirectGregorianInstantiation : Detector(), SourceCodeScanner {
...
     }
 
-    override fun getApplicableConstructorTypes(): List<String>? {
-        val forbiddenConstructors: MutableList<String> = ArrayList()
+    override fun getApplicableConstructorTypes(): List<String> {
+        val forbiddenConstructors: MutableList<String> = mutableListOf()
         forbiddenConstructors.add("java.util.GregorianCalendar")
         return forbiddenConstructors
```
Comment (david-allison) [inline lint-rules/src/main/java/com/ichi2/anki/lint/rules/DirectGregorianInstantiation.kt:49]:
```suggestion
        return mutableListOf("java.util.GregorianCalendar")
```

**Review by david-allison — CHANGES_REQUESTED** [review]

File: lint-rules/src/main/java/com/ichi2/anki/lint/rules/DirectGregorianInstantiation.kt [hunk 3]
```diff
@@ -1,9 +1,10 @@
+@file:Suppress("UnstableApiUsage", "KDocUnresolvedReference")
```
Comment (david-allison) [inline lint-rules/src/main/java/com/ichi2/anki/lint/rules/DirectGregorianInstantiation.kt:1]: Fix `KDocUnresolvedReference`

**Review by david-allison — CHANGES_REQUESTED** [review]

**Review by david-allison — APPROVED** [review]

**Pull request merged**